lib/components/viewers/related-stops-panel.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react'
+
+import {
+  getModeFromRoute,
+  getRouteIdForPattern,
+  getStopName,
+  getStopTimesByPattern,
+  routeIsValid
+} from '../../util/viewer'
+import Icon from '../narrative/mode-icon'
+import { ComponentContext } from '../../util/contexts'
+
+import NextArrivalForPattern from './next-arrival-for-pattern'
+import RelatedPanel from './related-panel'
+import ViewStopButton from './view-stop-button'
+
+class RelatedStopsPanel extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      expanded: false
+    }
+  }
+
+  static contextType = ComponentContext
+
+  _toggleExpandedView = () => {
+    this.setState({ expanded: !this.state.expanded })
+  }
+
+  _getStopIcon = (stop) => {
+    const { ModeIcon } = this.context
+    const modes = [...new Set(stop.routes?.map(getModeFromRoute))]
+    if (modes.length === 1) {
+      return <ModeIcon height={22} mode={modes[0]} width={22} />
+    }
+    return <Icon type='map-marker' />
+  }
+
+  render () {
+    const { expanded } = this.state
+    const {
+      homeTimezone,
+      nearbyStops,
+      stopViewerArriving,
+      stopViewerConfig,
+      timeFormat
+    } = this.props
+    return (
+      <RelatedPanel
+        count={nearbyStops?.length}
+        expanded={expanded}
+        onToggleExpanded={this._toggleExpandedView}
+        title='Related Stops'
+      >
+        <ul className='related-items-list list-unstyled'>
+          {nearbyStops?.map((stop, index) => {
+            const { id } = stop
+            if (!expanded && index >= 2) return null
+            const stopTimesByPattern = getStopTimesByPattern(stop)
+            const label = getStopName(stop)
+
+            return (
+              <li className='related-item' key={id}>
+                <div className='stop-label'>
+                  <div className='stop-name overflow-ellipsis' title={label}>
+                    <span className='child-stop-icon'>
+                      {this._getStopIcon(stop)}
+                    </span>
+                    <span className='stop-name-text'>{label}</span>
```

Comment:
This label is sometimes shrunk down. I would allow multi-line titles. 

<img width="417" alt="Screen Shot 2021-09-14 at 11 32 40 AM" src="https://user-images.githubusercontent.com/86619099/133242343-61f6f991-cc8f-45a9-9afa-a5d50b846a49.png">